### PR TITLE
fix: a11y fail   click events have key events and no static element interactions 

### LINKED ISSUE #4130: enhancement: a11y rules "click-events-have-key-events" and "no-static-element-interactions" fail with a warning.
### Describe the bug

[jsx-a11y eslint plugin](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) was recently added to the project and these rules -jsx-a11y/click-events-have-key-events and jsx-a11y/no-static-element-interactions - fail with a warning.

This issue offers more context: https://github.com/open-sauced/app/issues/2264

### Steps to reproduce

The following components and files are causing the rule to fail:

1. components/atoms/PillSelector/pill-selector.tsx
2. components/atoms/Radio/radio.tsx
3. components/atoms/RadioCheck/radio-check.tsx
4. components/atoms/SVGIcon/svg-icon.tsx
5. components/atoms/Search/search.tsx
6. components/atoms/TextInput/text-input.tsx
7. components/atoms/ToggleOption/toggle-option.tsx
8. components/molecules/AuthSection/auth-section.tsx
9. components/molecules/CardRepoList/card-repo-list.tsx
10. components/molecules/ContributorFilterDropdown/contributor-filter-dropdown.tsx
11. components/molecules/ContributorHighlight/contributor-highlight-card.tsx
12. components/molecules/ContributorListTableRow/contributor-list-table-row.tsx
13. components/molecules/DevCard/dev-card.tsx
14. components/molecules/HighlightInput/highlight-input-form.tsx
15. components/molecules/HighlightsFeedCard/highlights-filter-card.tsx
16. components/molecules/InfoCard/info-card.tsx
17. components/molecules/RepoRow/repo-row.tsx
18. components/molecules/ReposoitoryCartItem/repository-cart-item.tsx
19. components/molecules/SuperlativeSelector/superlative-selector.tsx
20. components/organisms/DevCardWall/dev-card-wall.tsx
21. components/organisms/SearchDialog/search-dialog.tsx
22. pages/u/[username]/card.tsx

### PR DESCRIPTION
**Pull request by FatumaA**
## Description
 Addresses the accessibility problems in the files listed in the linked issue.

## Related Tickets & Documents
Fixes #4130 

## Mobile & Desktop Screenshots/Recordings

**AuthSection:** 
authsection - after
authsection -before

**CardRepoList:**
cardrepolist - after
cardrepolist - before

**ContributorFilterDropdown:**
contributorFilterDropdown - after
contributorFilterDropdwon - before

**ContributorHighLightCard:**
Neutral and hiver states remain same - 
contributorHighlightCard

contributorHighloghtCard - hover

**ContributorlistTableRow:** 
No diff, except it now has a cursor on hover: 
contributorlistTableRow - no diff except cursor

**DevCard:**
devcard - no diff

**HighlightFilterCard:**
highlight filtercard - no diff

**HighlightInputForm close button:**
highlightinputform - close button

**HighlightInputForm - suggested links:**
highlightinputform - suggested highlight link

**PillSelector:**
pill selector

pill selector after

**RepositoryCartItem:**
repositorycartitem - no diff

**Search:**
search
search - focussed

**Search-dialog:**
search-dialog - no diff

**SuperlativeSelector:** 
Unselected - 
superlative selector - no diff
Selected - 
superlative selctor - selectec - no diff

**TextInput:**
textinput - before

textinput- after

**ToggleOption:**
toggleoption - after
toggleoption - before

With Icon
toggleoption with icon - after
toggleoption with icon - before

**Card in  pages/u/[username]/Card.tsx:**
u:username card - no diff

Radio and Radio Check: 
Screenshot 2024-10-23 at 01 02 01

Screenshot 2024-10-23 at 01 03 37

## Steps to QA
Run `npx eslint .` in the root folder, and see that those errors are no longer occurring

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?

A picture of a black woman chewing with her eyes darting nervously

### CODE REVIEW

File: components/atoms/Radio/radio.tsx [hunk 1]
```diff
@@ -16,6 +16,14 @@ const Radio = ({ className, withLabel, id, children, value, checked = false, onC
   return (
     <div
```
Comment (nickytonline) [inline components/atoms/Radio/radio.tsx:17]: This should be an actual button, and we can remove the `tabIndex` and `role="button"`

File: components/atoms/RadioCheck/radio-check.tsx [hunk 2]
```diff
@@ -14,6 +14,14 @@ const RadioCheck = ({ className, id, children, value, checked = false, onClick }
   return (
     <div
       onClick={onClick}
+      onKeyDown={(e) => {
```
Comment (nickytonline) [inline components/atoms/RadioCheck/radio-check.tsx:17]: This should be an actual button, and we can remove the `tabIndex` and `role="button"`

File: components/atoms/SVGIcon/svg-icon.tsx [hunk 3]
```diff
@@ -11,17 +11,26 @@ interface IconProps {
...
-        <use
-          href={IconImage}
-          width={size}
-          height={size}
-          className="items-center justify-center"
-          style={{ stroke: "currentcolor", maxWidth: "100%", height: "auto" }}
-        />
-      </svg>
-    </div>
+    <svg
```
Comment (nickytonline) [inline components/atoms/SVGIcon/svg-icon.tsx:14]: This should be an actual button that houses the SVG, and we can remove the `tabIndex`, `onKeyDown`, and `role="button" from the SVG and promote the `onClick` to the button.

File: components/atoms/Search/search.tsx [hunk 4]
```diff
@@ -150,10 +149,19 @@ const Search = ({
                 style={suggestionsStyle}
                 key={index}
                 data-suggestion={typeof suggestion === "string" ? suggestion : suggestion.key}
+                role="button"
```
Comment (nickytonline) [inline components/atoms/Search/search.tsx:152]: This should be an actual button, and we can remove the `tabIndex`, `onKeyDown` and `role="button"`

File: components/atoms/ToggleOption/toggle-option.tsx [hunk 5]
```diff
@@ -10,6 +10,15 @@ const ToggleOption = ({ optionText, withIcon, checked, handleToggle }: ToogleOpt
   return (
     <div
```
Comment (nickytonline) [inline components/atoms/ToggleOption/toggle-option.tsx:11]: The same comment as the other ones abou div to buttons and similar situations in the rest of the files in the PR.

**Review by nickytonline — CHANGES_REQUESTED** [review]
There are divs that need to converted to buttons, but we need to take focus states and styling into account. First convert the divs to buttons and then we can address the styling and focus states.

File: components/atoms/SVGIcon/svg-icon.tsx [hunk 6]
```diff
@@ -11,8 +11,8 @@ interface IconProps {
 
 const SVGIcon: React.FC<IconProps> = ({ onClick, className, IconImage, size = 16 }) => {
   return (
-    <div className={clsx(`inline-flex`, className)} onClick={onClick}>
-      <svg>
+    <button onClick={onClick}>
+      <svg className={clsx(`inline-flex`, className)}>
```
Comment (nickytonline) [inline components/atoms/SVGIcon/svg-icon.tsx:15]: Refactor (blocking): The CSS classes should be on the button as they were on the div previously.

```suggestion
    <button className={clsx(`inline-flex`, className)} onClick={onClick}>
      <svg>
```

File: components/molecules/CardRepoList/card-repo-list.tsx [hunk 7]
```diff
@@ -89,9 +87,9 @@ const CardRepoList = ({
                   ) : (
                     ""
                   )}
-                </div>
+                </button>
               );
-            })}
+            })}{" "}
```
Comment (nickytonline) [inline components/molecules/CardRepoList/card-repo-list.tsx:92]: Refactor (blocking): A blank space text node got added that shouldn't bet there.

```suggestion
            })}
```

File: components/molecules/SuperlativeSelector/superlative-selector.tsx [hunk 8]
```diff
@@ -50,23 +50,24 @@ const SuperativeSelector: React.FC<SuperlativeSelectorProps> = ({
...
+                  IconImage={cancelIcon}
+                  onClick={() => {
+                    handleCancelClick();
+                    setIsOpen(false);
+                  }}
+                />
+              </button>
             </div>
-          </div>
+          </>
```
Comment (nickytonline) [inline components/molecules/SuperlativeSelector/superlative-selector.tsx:68]: Refactor (blocking): The React Fragment is unnecessary as the code block for when `selected` is true only has one React.Node.

```suggestion
            <div className="flex">
              <span className="inline-block text-dark-slate-10">Filtered by:</span>
              <button className="block truncate pr-4 ml-1 text-light-slate-12" onClick={toggleFilter}>
                {filterDescription}
                <Icon
                  className="absolute right-1 top-1/2 transform -translate-y-1/2"
                  IconImage={cancelIcon}
                  onClick={() => {
                    handleCancelClick();
                    setIsOpen(false);
                  }}
                />
              </button>
            </div>
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
A few more change requests. Can you provide before and after screen shots of where these changes are?

In the case where it's a component that is used in multiple components, just have a before and after screenshots for a single place it's used.

Thanks @FatumaA and looking forward to getting these changes into the code base to make the app more accessible!

**Comment by FatumaA** [comment]
> A few more change requests. Can you provide before and after screen shots of where these changes are?
> 
> In the case where it's a component that is used in multiple components, just have a before and after screenshots for a single place it's used.
> 
> Thanks @FatumaA and looking forward to getting these changes into the code base to make the app more accessible!

Can I do this from the storybook instance? That's what I was using

**Comment by nickytonline** [comment]
> > A few more change requests. Can you provide before and after screen shots of where these changes are?
> > In the case where it's a component that is used in multiple components, just have a before and after screenshots for a single place it's used.
> > Thanks @FatumaA and looking forward to getting these changes into the code base to make the app more accessible!
> 
> Can I do this from the storybook instance? That's what I was using

Yeah, just point to the Storybook stories.

**Comment by nickytonline** [comment]
Looks like there are some conflicts that need to be resolved @FatumaA

**Comment by FatumaA** [comment]
As for the other components, I did not notice any difference in their look or function with the changes

**AuthSection:** 
authsection - after
authsection -before

**CardRepoList:**
cardrepolist - after
cardrepolist - before

**ContributorFilterDropdown:**
contributorFilterDropdown - after

**ContributorHighLightCard:**
Neutral and hiver states remain same - 
contributorHighlightCard

contributorFilterDropdwon - before
contributorHighloghtCard - hover

**ContributorlistTableRow:** 
No diff, except it now has a cursor on hover: 
contributorlistTableRow - no diff except cursor

**DevCard:**
devcard - no diff

**HighlightFilterCard:**
highlight filtercard - no diff

**HighlightInputForm close button:**
highlightinputform - close button

**HighlightInputForm - suggested links:**
highlightinputform - suggested highlight link

**PillSelector:**
pill selector

pill selector after

**RepositoryCartItem:**
repositorycartitem - no diff

**Search:**
search
search - focussed

**Search-dialog:**
search-dialog - no diff

**SuperlativeSelector:** 
Unselected - 
superlative selector - no diff
Selected - 
superlative selctor - selectec - no diff

**TextInput:**
textinput - before

textinput- after

**ToggleOption:**
toggleoption - after
toggleoption - before

With Icon
toggleoption with icon - after
toggleoption with icon - before

**Card in  pages/u/[username]/Card.tsx:**
u:username card - no diff

**Comment by nickytonline** [comment]
Thanks for all the screenshots for before and after @FatumaA. So we can move this along, can you exclude changes that currently break the UI and we can sort those out in a follow up PR?

Also, thanks for your patience!

**Comment by FatumaA** [comment]
> Thanks for all the screenshots for before and after @FatumaA. So we can move this along, can you exclude changes that currently break the UI and we can sort those out in a follow up PR?
> 
> Also, thanks for your patience!

No problem, thank you for yours! 

I think the radio and radio check should be a simple fix on the button classes, we should keep the fix in this PR; what do you think?

As for the devwallcard component, I will undo the changes and leave it out as you've suggested.

**Comment by nickytonline** [comment]
For the radio and radio check feel free to put the fix in here, but also, if you want to take a bit of time, like I said, it can go in a follow up PR.

**Comment by FatumaA** [comment]
Fixed the radio and radio check. And undid the devcardwall change

Radio and Radio Check: 
Screenshot 2024-10-23 at 01 02 01

Screenshot 2024-10-23 at 01 03 37

**Comment by nickytonline** [comment]
This seems to all be working great. The only thing I noticed is the search no longer auto focuses when you click on the search button or press CMD + K (macOS) or CTRL + K (other OSes).

**Comment by FatumaA** [comment]
> This seems to all be working great. The only thing I noticed is the search no longer auto focuses when you click on the search button or press CMD + K (macOS) or CTRL + K (other OSes).

I removed the auto focus on the text input component, let me look into it

**Comment by FatumaA** [comment]
@nickytonline I pushed a fix for that, but I noticed the component uses global document methods. Is there any particular reason, or is this up for a refactor?

**Comment by nickytonline** [comment]
@FatumaA, I'll take a peek at this tomorrow.

**Review by nickytonline — APPROVED** [review]
Thanks for this @FatumaA! 🚢

**Review by zeucapua — APPROVED** [review]

File: .env [hunk 9]
```diff
@@ -4,6 +4,8 @@ NEXT_PUBLIC_API_URL=https://beta.api.opensauced.pizza/v2
 NEXT_PUBLIC_EXP_API_URL=https://beta.api.opensauced.pizza/v2
 NEXT_PUBLIC_POSTHOG_ID=phc_Y0xz6nK55MEwWjobJsI2P8rsiomZJ6eZLoXehmMy9tt
 NEXT_PUBLIC_CLOUD_NAME=dgxgziswe
+NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY=test
```
Comment (brandonroberts) [inline .env:7]: This change should be reverted

File: .env [hunk 10]
```diff
@@ -4,6 +4,8 @@ NEXT_PUBLIC_API_URL=https://beta.api.opensauced.pizza/v2
...
+NEXT_PUBLIC_STRIPE_SUB_CANCEL_URL=https://billing.stripe.com/p/login/test_bIY8xy3iy6fpeHu3cc
```
Comment (brandonroberts) [inline .env:8]:
```suggestion
```

**Review by brandonroberts — APPROVED** [review]
Minor nit about the .env

**Pull request merged**